source/shadow.c

```diff
@@ -543,7 +543,7 @@ ShadowStatus_t Shadow_GetTopicString( ShadowTopicStringType_t topicType,
     uint16_t offset = 0U, generatedTopicStringLength = 0U, operationStringLength = 0U;
     ShadowStatus_t shadowStatus = SHADOW_SUCCESS;
     const char * pOperationString = NULL;
-    const char shadowPrefix[ SHADOW_PREFIX_LENGTH ] = SHADOW_PREFIX;
+    const char shadowPrefix[ SHADOW_PREFIX_LENGTH + 1 ] = SHADOW_PREFIX;
```

Comment:
Why make a copy on stack? Why not `const char * shadowPrefix = SHADOW_PREFIX`?

Reply:
It seems like the string is copied into the topic buffer that is generated by this function. It can just use `SHADOW_PREFIX` instead of storing a local reference to it.

Reply:
That is needed to silence MISRA violation.

Reply:
Thanks for all your suggestion! I'll take ```const char * shadowPrefix = SHADOW_PREFIX``` as the solution, and it pass the MISRA as well.